graphql/schemabuilder/pagination.go

```diff
@@ -420,15 +518,41 @@ func (sb *schemaBuilder) buildPaginatedField(typ reflect.Type, f interface{}) (*
 	return ret, nil
 }
 
-func (funcCtx *funcContext) extractPaginatedRetAndErr(nodeKey string, out []reflect.Value, args interface{}, retType graphql.Type) (interface{}, error) {
+func (funcCtx *funcContext) extractPaginatedRetAndErr(nodeKey string, out []reflect.Value, args interface{}, retType graphql.Type, embedsArgs bool, returnsConnInfo bool) (interface{}, error) {
 	var result interface{}
-	connectionArgs, _ := args.(ConnectionArgs)
+	var paginationArgs PaginationArgs
+
+	// If the pagination args are not embedded then they need to be extracted out of ConnectionArgs
+	// struct and setup for the slicing functions.
+	if !embedsArgs {
+		connectionArgs, _ := args.(ConnectionArgs)
+		paginationArgs = PaginationArgs{
+			First:  connectionArgs.First,
+			Last:   connectionArgs.Last,
+			After:  connectionArgs.After,
+			Before: connectionArgs.Before,
+		}
+	} else {
+		pagTyp := reflect.TypeOf(args)
+		fieldInd := -1 // will panic
+		for i := 0; i < pagTyp.NumField(); i++ {
+			field := pagTyp.Field(i)
+			if field.Type == reflect.TypeOf(PaginationArgs{}) {
+				fieldInd = i
+			}
+		}
+
+		paginationArgs, _ = reflect.ValueOf(args).Field(fieldInd).Interface().(PaginationArgs)
```

Comment:
second return value is only necessary if you're going to use it

Reply:
gotcha